File: 1498-number-of-subsequences-that-satisfy-the-given-sum-condition/1498-number-of-subsequences-that-satisfy-the-given-sum-condition.py
class Solution:
    def numSubseq(self, nums: List[int], target: int) -> int:
        nums.sort()
        i=0
        j=len(nums)-1
        ans=0
        mod=pow(10,9)+7
        while i<=j:
            if nums[i]+nums[j]>target:
                j-=1
            else:
                ans+=pow(2,j-i,mod)
                i+=1
        return ans%mod

File: 1498-number-of-subsequences-that-satisfy-the-given-sum-condition/1498-number-of-subsequences-that-satisfy-the-given-sum-condition.java
class Solution {
    public int numSubseq(int[] nums, int target) {
        Arrays.sort(nums);
        int i = 0, j = nums.length - 1;
        long ans = 0;
        int mod = (int)(Math.pow(10, 9)) + 7;
        int[] powers = new int[nums.length];
        powers[0] = 1;
        for (int k = 1; k < nums.length; k++)
            powers[k] = (powers[k - 1] * 2) % mod;
        while (i <= j) {
            if (nums[i] + nums[j] > target) {
                j--;
            }
            else {
                // System.out.println(ans + " " + i + " " + j);
                ans += powers[j - i];
                ans %= mod;
                i++;
            }
        }
        return (int) ans;
    }
}